src/main/java/com/edmunds/rest/databricks/service/ClusterServiceImpl.java

```diff
@@ -125,4 +153,30 @@ public ClusterInfoDTO getInfo(String clusterId) throws IOException, DatabricksRe
     ClusterEventsDTO clusterEvents = this.mapper.readValue(responseBody, ClusterEventsDTO.class);
     return clusterEvents.getEvents();
   }
+
+  @Override
+  public void upsertCluster(NewClusterDTO clusterDTO) throws IOException, DatabricksRestException {
+    boolean clusterExists = false;
+    String clusterId = "";
+    ClusterInfoDTO[] clusters = list();
+    String clusterName = clusterDTO.getClusterName();
+    for (ClusterInfoDTO cluster : clusters) {
```

Comment:
How about refactor this lookup to its own method and add to interface? its a useful operation.
I don't believe you can have a case with multiple *running* clusters with same name... but that would be good to verify. For jobs for example, there is different logic depending on what people want.

Reply:
Done, for multiple clusters I've chosen error logging + updating first one